### PR TITLE
more parliament permissions

### LINKED ISSUE #2118: Parliament Users
Parliament should use Arkime users to authenticate users.

Non parliament user:

- [x] view only

Parliament user
- [x] dismiss issues

Parliament admin:
- [x] dismiss issues
- [x] configure parliament
- [x] update parliament settings

Settings:
- [ ] Add http realm

### PR DESCRIPTION
**Pull request by 31453**
hide login button if not a parliament user
hide edit parliament button if not a parliament admin show cluster/group errors fixed on the bottom for visibility

fixes #2118

### CODE REVIEW

File: parliament/parliament.js [hunk 1]
```diff
@@ -1232,10 +1236,6 @@ router.put('/auth/commonauth', [checkAuthUpdate], (req, res, next) => {
     return next(newError(403, 'Your Parliament is in dasboard only mode. You cannot setup auth.'));
   }
 
-  if (!ArkimeUtil.isString(req.body.commonAuth)) {
```
Comment (awick) [inline parliament/parliament.js:1235]: should still check if it is !== undefined?

**Review by awick — CHANGES_REQUESTED** [review]

File: parliament/vueapp/src/components/Settings.vue [hunk 2]
```diff
@@ -434,6 +434,24 @@
...
+                  </span>
+                </span>
+                <input type="string"
+                  class="form-control"
+                  id="httpRealm"
+                  v-model="settings.commonAuth.httpRealm"
+                />
+              </div>
+              <p class="form-text small text-muted">
+                The HTTP Realm for user authentication. Set this to the same realm as your Akime clusters to share credentials. By default "Moloch" if not set.
```
Comment (31453) [inline parliament/vueapp/src/components/Settings.vue:452]: @awick Want to help me with this text. It sounds awkward to me 🤷

File: tests/parliament.t [hunk 3]
```diff
@@ -160,3 +160,194 @@ eq_or_diff($result, from_json('{"tokenError":true,"success":false,"text":"Permis
...
+$result = parliamentPostToken2("/parliament/api/groups/0/clusters", '{"title":"Cluster 1","url":"localhost:8123"}', $parliamentAdminToken);
+ok($result->{success});
+$result = parliamentPutToken2("/parliament/api/groups/0", '{"title":"Group 1a"}', $parliamentAdminToken);
+ok($result->{success});
+$result = parliamentPutToken2("/parliament/api/groups/0/clusters/0", '{"url":"localhost:8123","title":"Cluster 1a"}', $parliamentAdminToken);
+ok($result->{success});
+$result = parliamentDeleteToken2("/parliament/api/groups/0/clusters/0", $parliamentAdminToken);
+ok($result->{success});
+$result = parliamentDeleteToken2("/parliament/api/groups/0", $parliamentAdminToken);
+ok($result->{success});
```
Comment (awick) [inline tests/parliament.t:353]: Can you add ```viewerGet("/regressionTests/deleteAllUsers");``` to end

File: tests/parliament.t [hunk 4]
```diff
@@ -160,3 +160,194 @@ eq_or_diff($result, from_json('{"tokenError":true,"success":false,"text":"Permis
...
+$result = parliamentPutToken2("/parliament/api/removeIgnoreIssues", '{}', $parliamentUserToken);
+eq_or_diff($result, from_json('{"text": "Must specify the issue(s) to unignore.", "success": false}'));
+$result = parliamentPutToken2("/parliament/api/groups/0/clusters/0/removeIssue", '{}', $parliamentUserToken);
+eq_or_diff($result, from_json('{"text": "Must specify the issue type to remove.", "success": false}'));
+$result = parliamentPutToken2("/parliament/api/issues/removeAllAcknowledgedIssues", '{}', $parliamentUserToken);
+eq_or_diff($result, from_json('{"text": "There are no acknowledged issues to remove.", "success": false}'));
+$result = parliamentPutToken2("/parliament/api/removeSelectedAcknowledgedIssues", '{}', $parliamentUserToken);
+eq_or_diff($result, from_json('{"text": "Must specify the acknowledged issue(s) to remove.", "success": false}'));
+
+# parliament user cannot access/udpate settings/parliament
```
Comment (awick) [inline tests/parliament.t:258]: would be nice if at least some of these tested against the text ;)

**Pull request merged**